submit_cgap/submission.py

```diff
@@ -625,30 +632,138 @@ def do_uploads(upload_spec_list, auth, folder=None, no_query=False, subfolders=F
     if subfolders:
         folder = os.path.join(folder, '**')
     for upload_spec in upload_spec_list:
-        file_path = os.path.join(folder, upload_spec['filename'])
-        file_search = glob.glob(file_path, recursive=subfolders)
-        if len(file_search) == 1:
-            filename = file_search[0]
-        elif len(file_search) > 1:
-            show(
-                "No upload attempted for file %s because multiple copies were found"
-                " in folder %s: %s."
-                % (upload_spec['filename'], folder, ", ".join(file_search))
-            )
+        file_metadata = None
+        file_name = upload_spec["filename"]
+        file_path, error_msg = search_for_file(folder, file_name, recursive=subfolders)
+        if error_msg:
+            show(error_msg)
             continue
-        else:
-            filename = file_path
         uuid = upload_spec['uuid']
-        if not no_query:
-            if CGAP_SELECTIVE_UPLOADS and not yes_or_no("Upload %s?" % (filename,)):
-                show("OK, not uploading it.")
-                continue
-        try:
-            show("Uploading %s to item %s ..." % (filename, uuid))
-            upload_file_to_uuid(filename=filename, uuid=uuid, auth=auth)
-            show("Upload of %s to item %s was successful." % (filename, uuid))
-        except Exception as e:
-            show("%s: %s" % (e.__class__.__name__, e))
+        uploader_wrapper = UploadMessageWrapper(uuid, no_query=no_query)
+        wrapped_upload_file_to_uuid = uploader_wrapper.wrap_upload_function(
+            upload_file_to_uuid, file_path
+        )
+        file_metadata = wrapped_upload_file_to_uuid(
+            filename=file_path, uuid=uuid, auth=auth
+        )
+        if file_metadata:
+            extra_files_credentials = file_metadata.get("extra_files_creds", [])
+            if extra_files_credentials:
+                upload_extra_files(
+                    extra_files_credentials,
+                    uploader_wrapper,
+                    folder,
+                    auth,
+                    recursive=subfolders,
+                )
+
```

Comment:
Should there be an `else` here that raises an error or otherwise makes some noise about having file metadata but no credentials? Weird to be silent here.